cpp/include/raft/util/inline.hpp

```diff
@@ -16,6 +16,20 @@
 
 #pragma once
 
+// The RAFT_INLINE_CONDITIONAL is a conditional inline specifier that removes
```

Comment:
Good idea. I have defined `RAFT_WEAK_FUNCTION` in `macros.hpp` and documented it there. It does not need to be conditional imho because the behavior is correct regardless of `RAFT_COMPILED`.